src/internal_events/loki.rs

```diff
@@ -71,3 +71,18 @@ impl InternalEvent for LokiOutOfOrderEventRewritten {
         counter!("rewritten_timestamp_events_total", self.count as u64);
     }
 }
+
+#[derive(Debug)]
+pub struct LokiEventTimestampOutOfRangeError;
+
+impl InternalEvent for LokiEventTimestampOutOfRangeError {
+    fn emit(self) {
+        error!(
```

Comment:
Should we make this a `warn!` since it doesn't actually inhibit processing? Also, what happens when we send an event to Loki without a timestamp? Does it just use the current timestamp?

Reply:
According to Loki api, timestamp is  must.

Reply:
Oh I see, I missed that you are returning early. In that case we'd want to emit the metrics for errors and discarded events. See an example here: https://github.com/vectordotdev/vector/blob/8325300a767bf342f455ad796fe01c9722bb8fd9/src/internal_events/splunk_hec.rs#L55-L77